vortex-array/src/array/varbin/builder.rs

```diff
@@ -47,9 +46,9 @@ impl<O: NativePType> VarBinBuilder<O> {
         self.validity.append_null();
     }
 
-    pub fn finish(&mut self, dtype: DType) -> VarBinArray {
-        let offsets = PrimitiveArray::from(mem::take(&mut self.offsets));
-        let data = PrimitiveArray::from(mem::take(&mut self.data));
+    pub fn finish(mut self, dtype: DType) -> VarBinArray {
+        let offsets = PrimitiveArray::from(self.offsets);
+        let data = PrimitiveArray::from(Vec::from(self.data.freeze()));
```

Comment:
This isn't zero copy is it? We can add a function to PrimitiveArray to construct directly from a buffer (for PType==u8?)

Reply:
Alternatively, @robert3005 did we discuss VarBin data being a buffer instead of a child array?

Reply:
I think we thought about it but then we wanted to have something like ZstdEncoding

Reply:
I’m starting to think general purpose compression can be configured on buffers at write-time though; using the layouts mechanism.

Reply:
Not zero-copy, but still pretty cheap IMO. Constructing things from `Bytes` is hard because there's no guarantee the instance is exclusive

Reply:
But PrimitiveArray wraps a vortex-buffer, which itself wraps Bytes. So this copy is purely because the right API isn't exposed / isn't used

Reply:
true, fixing